### ISSUE TITLE
🙏  Help Wanted: Documentation for New Backend System Everywhere

### ISSUE THREAD

**Issue by awanlin** [backstage/backstage#24195]
**TL;DR** With the New Backend System being the default when you create a new Backstage instance we need to update all the related backend documentation with proper examples and guidance. This will also help out existing Adopters who are migrating as there are still more advanced cases that haven't been covered yet.

We need help getting all the documentation up to date so that new and existing Adopters can find the information they need!

## How do I help?

- The process is to identify an area that needs to be updated and provide the needed documentation for it;
- Please add `[NBS Docs]` to the tittle of your PR so that way we have an easy way to identify it;
- Crosslink the outdated page with the updated page - make a copy of the outdated page, add the `--old` prefix to its filename and metadata id (see the completed pull requests linked in this issue as examples).

The next section will list out areas that need work, please check for any active PRs before starting.

## Pages needing updates

The following is a list of pages that need to be updated (as of 2024-08-26):

> [!Note]
> If there is something missing from this list please feel free to mention it or even better submit a PR with the updates and mention it here 👍 

### Plugins
```[tasklist]
- [ ] https://github.com/backstage/backstage/issues/26239
- [ ] https://github.com/backstage/backstage/pull/26238
```

### Auth
Most of these just need to add details about the `signIn.resolvers`.
```[tasklist]
- [ ] https://github.com/backstage/backstage/pull/26379
- [ ] https://github.com/backstage/backstage/issues/26395
- [ ] https://github.com/backstage/backstage/issues/26408
- [ ] https://github.com/backstage/backstage/issues/26403
```

### Permissions
```[tasklist]
- [ ] https://github.com/backstage/backstage/issues/26279
- [ ] https://github.com/backstage/backstage/issues/26334
- [ ] https://github.com/backstage/backstage/issues/26364
- [ ] https://github.com/backstage/backstage/issues/26376
```

### Completed:

- Core Features:
  - [x] https://backstage.io/docs/features/software-catalog/extending-the-model
  - [x] https://backstage.io/docs/features/search/getting-started
  - [x] https://backstage.io/docs/features/search/search-engines
  - [x] https://backstage.io/docs/features/search/how-to-guides

- Integrations:
  - [x] https://backstage.io/docs/integrations/aws-s3/discovery
  - [x] https://backstage.io/docs/integrations/azure/discovery - https://github.com/backstage/backstage/pull/24222
  - [x] https://backstage.io/docs/integrations/azure/org - https://github.com/backstage/backstage/pull/24222
  - [x] https://backstage.io/docs/integrations/bitbucketCloud/discovery - https://github.com/backstage/backstage/pull/22344
  - [x] https://backstage.io/docs/integrations/bitbucketServer/discovery
  - [x] https://backstage.io/docs/integrations/gerrit/discovery
  - [x] https://backstage.io/docs/integrations/github/discovery - https://github.com/backstage/backstage/pull/24573
  - [x] https://backstage.io/docs/integrations/github/org - https://github.com/backstage/backstage/pull/24573
  - [x] https://backstage.io/docs/integrations/gitlab/discovery - https://github.com/backstage/backstage/pull/23373
  - [x] https://backstage.io/docs/integrations/gitlab/org - https://github.com/backstage/backstage/pull/23373
  - [x] https://backstage.io/docs/integrations/ldap/org - #24909
- Plugins
  - [x] https://backstage.io/docs/plugins/proxying
  - [x] https://backstage.io/docs/plugins/backend-plugin (#24069)
  - [x] https://backstage.io/docs/plugins/observability (Health check)
- Auth (most of these just need to add details about the `signIn.resolvers`)
  - [x] https://backstage.io/docs/auth/atlassian/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/microsoft/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/microsoft/easy-auth - https://github.com/backstage/backstage/pull/23909
  - [x] https://backstage.io/docs/auth/bitbucket/provider
  - [x] https://backstage.io/docs/auth/cloudflare/provider - https://github.com/backstage/backstage/pull/23997
  - [x] https://backstage.io/docs/auth/github/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/gitlab/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/google/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/google/gcp-iap-auth - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/okta/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/oauth2-proxy/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/onelogin/provider
  - [x] https://backstage.io/docs/auth/vmware-cloud/provider - https://github.com/backstage/backstage/pull/24221
  - [x] https://backstage.io/docs/auth/service-to-service-auth - https://github.com/backstage/backstage/pull/24257
- Permissions
  - [x] https://backstage.io/docs/permissions/getting-started - https://github.com/backstage/backstage/pull/24229
  - [x] https://backstage.io/docs/permissions/writing-a-policy
  - [x] https://backstage.io/docs/permissions/custom-rules
- Tutorials
  - [x] https://backstage.io/docs/tutorials/setup-opentelemetry

**Comment by awanlin** [comment]
Thanks @elaine-mattos, I've updated the list with that PR 🚀

**Comment by awanlin** [comment]
Hi @Parsifal-M, I've been just adding on the new backend system examples, my thinking is that being able to seem them together could be helpful.

**Comment by freben** [comment]
I made a stab at making an `--old` file that's not in the sidebar but can be linked to.

https://github.com/backstage/backstage/pull/24154

**Comment by awanlin** [comment]
PR https://github.com/backstage/backstage/pull/24221 covers the auth pages for: Atlassian, GCP IAP, GitHub, GitLab, Google, Microsoft, oauth2 Proxy, Okta, and VMware Cloud 🚀

**Comment by awanlin** [comment]
I've asked that they update the `azure-easyauth` docs as part of this PR https://github.com/backstage/backstage/pull/23909

**Comment by awanlin** [comment]
I've asked that they update the `cloudflare` docs as part of this PR https://github.com/backstage/backstage/pull/23997

**Comment by awanlin** [comment]
PR https://github.com/backstage/backstage/pull/24222 covers the integration pages for Azure and MS Graph

**Comment by awanlin** [comment]
I've started work on the Service to Service auth page but need a bit of guidance on this - https://github.com/backstage/backstage/pull/24223

**Comment by ManuInNZ** [comment]
I can add  things for the GitHub Org data using GH app and multi orgs/apps eventually

**Comment by Parsifal-M** [comment]
#24229 started working on the permissions pages, also FYI @vinzscam :heart: LMK!

**Comment by razvanonet** [comment]
@ManuInNZ I'm really eager to learn how the GitHub Org data using GH app is set up using the new backend system. If you've done some progress on that in a PR or something, please put it here

**Comment by prasoon-pxc** [comment]
I am also looking doc of GithHub Org data using GH-App, as earlier instruction not working for new-backend system

**Comment by prasoon-pxc** [comment]
@awanlin , I have tried out the updated [doc](https://github.com/awanlin/backstage/blob/topic/nbs-integration-github/docs/integrations/github/org.md) for user&group ingestion, but it's not working for me, have not added any events-support and custom-transformer things

 I have also tried [discovery](https://github.com/awanlin/backstage/blob/topic/nbs-integration-github/docs/integrations/github/discovery.md) doc and it's working

**Comment by prasoon-pxc** [comment]
Hi @awanlin , I have setup Github-integration with github app using this [doc](https://github.com/awanlin/backstage/blob/topic/nbs-integration-github/docs/integrations/github/github-apps.md) also provide required permissions for Github-app to ingest users and teams

After that added below package:

```yarn --cwd packages/backend add @backstage/plugin-catalog-backend-module-github-org```

added below line in `packages/backend/src/index-tx`

```
backend.add(import('@backstage/plugin-catalog-backend-module-github-org'));
```

`app-config.yaml`

```
integrations:
  github:
    - host: github.com
      apps:
        - $include: my-big-org.yaml
```

**Comment by awanlin** [comment]
@prasoon-pxc can you please post this here: https://github.com/backstage/backstage/issues/24559

**Comment by riginoommen** [comment]
LDAP i am actively looking for it

**Comment by ManuInNZ** [comment]
> I've got a PR to update the GitHub Discovery and Org integration pages - #24573 - feel free to take a look.
> 
> @ManuInNZ, sorry, I know you posted that you would be working on this but we are seeing a lot of questions here and on Discord so I felt we needed to get moving. Feel free to review my PR, feedback would be very welcome.
> 
> @razvanonet @prasoon-pxc feel free to try out that PR, perhaps it can help you out, if not maybe add comments so I can improve it.

Sorry, I got side tracked on contributing to another project. Will review but I am sure it is great.

**Comment by awanlin** [comment]
Hi everyone, 20 out of the 48 pages have been updated or have a pending PR. We aren't even half way yet so if you can help out please do! 🙏

**Comment by aramissennyeydd** [comment]
@awanlin I think we can remove https://backstage.io/docs/auth/troubleshooting/ and https://backstage.io/docs/plugins/call-existing-api from the list. AFAICT they're both backend system agnostic - they definitely need a revamp but I think that can be a separate issue.

**Comment by awanlin** [comment]
Sure @aramissennyeydd, I've removed those two 👍

**Comment by fmartinezperez** [comment]
Hi @awanlin, I'm working on migration to new backend system, and I have problems with the auth-oidc. I started using legacyPlugin, but it didn't work. After I used @backstage/plugin-auth-backend-module-oidc-provider but not exists documentation and only with the import is not enough. 
Could you help me at this point? All the documentation related with OIDC is outdated.
Thanks.

**Comment by Maetis** [comment]
Hi @awanlin, I've submitted a PR (#25427) for the following page: [https://backstage.io/docs/integrations/aws-s3/discovery](https://backstage.io/docs/integrations/aws-s3/discovery).

**Comment by Maetis** [comment]
Hi @awanlin, I've submitted a new PR (#25535) for the following page: [https://backstage.io/docs/integrations/bitbucketServer/discovery](https://backstage.io/docs/integrations/bitbucketServer/discovery)

**Comment by dweber019** [comment]
@awanlin 

https://backstage.io/docs/integrations/ldap/org (I don't think the LDAP plugin has been migrated to the new backend system yet)

Has been done with https://github.com/backstage/backstage/pull/24995

**Comment by Maetis** [comment]
Hi @awanlin, I've submitted a new PR (#25832) for the following page: https://backstage.io/docs/integrations/gerrit/discovery

**Comment by Rugvip** [comment]
Pointing to #25930 from here too, since the k8s docs could use a further tweak to more prominently present instructions for the new backend system.

**Comment by zeshanziya** [comment]
Hi @awanlin, I have submitted a PR #26146 for https://github.com/backstage/backstage/tree/master/plugins/catalog-backend-module-incremental-ingestion

**Comment by jhaals** [comment]
@awanlin I looked at the top 4 links under core features and it looks like those are all mentioning then new backend system. Is there something else that also should go in there or can we consider those done?

**Comment by camilaibs** [comment]
Related: https://github.com/backstage/backstage/issues/26177

**Comment by awanlin** [comment]
@jhaals, I've let updating this slide, I've added it to my todo list for today to review them all. Thanks for the reminder 👍

**Comment by awanlin** [comment]
Alright, list has been updated, moved the outstanding items to the top to make them a bit more apparent.

I'd like to highlight the fact that the auth0 and Bitbucket Server auth providers need to be migrated before we can even write the docs for them. We'll need some Community help with that.

**Comment by camilaibs** [comment]
@awanlin I added my name to the https://backstage.io/docs/plugins/integrating-search-into-plugins, just started working on it

**Comment by awanlin** [comment]
Thanks for updating this @camilaibs with the latest state and for the various PR to help keep this moving forward! 🚀

**Comment by camilaibs** [comment]
> Thanks for updating this @camilaibs with the latest state and for the various PR to help keep this moving forward! 🚀

We are almost there, let's get it done 🎉

**Comment by awanlin** [comment]
Closing as from my review to the best of my abilities everything has been updated. If anything was missed we can spin up new issues for them. Thanks everyone for your help and support with this! 🚀
